executor/logger/worker.go

```diff
@@ -31,16 +31,29 @@ const (
 // NewWorker creates, and returns a new Worker object. Its only argument
 // is a channel that the worker can add itself to whenever it is done its
 // work.
-func NewWorker(id int, workQueue chan LogRequest, log logr.Logger, sdepName string, namespace string, predictorName string, kafkaBroker string, kafkaTopic string) (*Worker, error) {
+func NewWorker(id int, workQueue chan LogRequest, log logr.Logger, sdepName string, namespace string, predictorName string, kafkaBroker string, kafkaTopic string, sslKakfa SslKakfa) (*Worker, error) {
 
 	var producer *kafka.Producer
 	var err error
 	if kafkaBroker != "" {
 		log.Info("Creating producer", "broker", kafkaBroker, "topic", kafkaTopic)
-		producer, err = kafka.NewProducer(&kafka.ConfigMap{
-			"bootstrap.servers":   kafkaBroker,
+		var producerConfigMap = kafka.ConfigMap{"bootstrap.servers": kafkaBroker,
 			"go.delivery.reports": false, // Need this othewise will get memory leak
-		})
+		}
+		log.Info("kafkaSecurityProtocol", "kafkaSecurityProtocol", sslKakfa.kafkaSecurityProtocol)
+		if kafkaBroker != "" {
+			if sslKakfa.kafkaSecurityProtocol != "" {
```

Comment:
This should probably be an explicit check for `sslKafka.kafkaSecurityProtocol == "ssl"` as these details are irrelevant to SASL, for example

Reply:
Fair, in my mind we would be explicit if we wanted to use SASL but I guess it's because I had our case in mind and not really the rest. I'm gonna change it :D